MapboxCoreNavigationTests/FreeDriveLocationManager.swift

```diff
@@ -0,0 +1,92 @@
+import Foundation
+import MapboxNavigationNative
+import TestHelper
+import XCTest
+@testable import MapboxCoreNavigation
+import MapboxAccounts
+
+class Road {
+    let from: CLLocationCoordinate2D
+    let to: CLLocationCoordinate2D
+    let length: Double
+
+    init(from: CLLocationCoordinate2D, to: CLLocationCoordinate2D) {
+        self.from = from
+        self.to = to
+        self.length = (to - from).length()
+    }
+
+    func whichCloser(a: CLLocationCoordinate2D, b: CLLocationCoordinate2D) -> CLLocationCoordinate2D {
+        return proximity(of: a) > proximity(of: b) ? b : a
+    }
+
+    func proximity(of: CLLocationCoordinate2D) -> Double {
+        return ((of - from).length() + (of - to).length()) / length
+    }
+}
+
+extension CLLocationCoordinate2D {
+    static func -(a: CLLocationCoordinate2D, b: CLLocationCoordinate2D) -> CLLocationCoordinate2D {
+        return CLLocationCoordinate2D(latitude: a.latitude - b.latitude, longitude: a.longitude - b.longitude)
+    }
+
+    func length() -> Double {
+        return sqrt(latitude * latitude + longitude * longitude)
+    }
+}
+
+class DebugInfoListener: FreeDriveDebugInfoListener {
+    var onUpdated: ((CLLocationCoordinate2D, CLLocationCoordinate2D)->Void)?
+
+    init(onUpdated: ((CLLocationCoordinate2D, CLLocationCoordinate2D)->Void)? = nil) {
+        self.onUpdated = onUpdated
+    }
+
+    func didGet(location: CLLocation, with matches: [MapMatch], for rawLocation: CLLocation) {
+        onUpdated?(rawLocation.coordinate, location.coordinate)
+    }
+}
+
+class FreeDriveLocationManagerTests: XCTestCase {
+    class LocationsObserver: NSObject, CLLocationManagerDelegate {
+        func locationManager(_ manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
+            guard let location = locations.first else { return }
+            print("Got location: \(location.coordinate.latitude), \(location.coordinate.longitude)")
+        }
+    }
+
+    func testFreeDrive() {
+        let locationManager = FreeDriveLocationManager()
```

Comment:
This test doesn’t really exercise free driving, other than the ability to set a custom location and get it back out of the location manager. `RouteTilesVersion.directionsCredentials` has a nil `host`, making `RouteTilesVersion.requestVersionsURL` return nil because of an invalid URL, causing `RouteTilesVersion.getAvailableVersions(completionHandlerQueue:completion:)` to bail early before ever kicking off a network request. Not that we’d want a unit test to kick off a network request, but because RouteTilesVersion is unable to return a version, `FreeDriveLocationManager.createNavigator(withTilesVersion:)` never configures the navigator.

Reply:
For now, I’ve changed the test to manually force the location manager to configure the navigator, but that’s quite contrived and won’t catch any failure modes.

Reply:
This is apparently the cause of the test failure on CI, where we haven’t configured a .mapbox file that the build system would read by default.